addons/zeus/functions/fnc_moduleGroupSide.sqf

```diff
@@ -25,20 +25,30 @@ if (_side == _newSide) exitWith {};
 private _oldGroup = group _unit;
 private _newGroup = createGroup _newSide;
 
+// Preserve groupid from the previous group if doesn't already exist
+if ((allGroups findIf {side _x isEqualTo _newSide && {(groupId _oldGroup) isEqualTo (groupId _newGroup)}}) == -1) then {
+    _newGroup setGroupIdGlobal [groupId _oldGroup];
+};
```

Comment:
Couldn't you just use `setGroupIdGlobal` without check, and if it is already taken, it would just fail silently anyway?

Reply:
I just thought it would be better to avoid potentially undefined behavior, but I guess I can test it to make sure nothing odd happens and then get rid of the check.

Reply:
I wonder if these even have to be unique. I would do the check if:
- the game bugs out with duplicated ids
- the game complains about it in RPT, but still changes the id
I would maybe do the check if, depending on the time of day if:
- the game complains about it in RPT, but doesn't change the id

Reply:
I'm not exactly sure what to do here. Here is what I discovered:
- You can have multiple groups with the same ID if they are on different sides
- You can not have multiple groups with the same ID if they are on the same side
- If you try to set a group id to something already taken on that side, they switch IDs

This means the following is possible:
 - BLUFOR has a group "Alpha-1"
 - OPFOR has a group "Alpha-1"
 - The Group Side module is used on the BLUFOR group and set to OPFOR
 - The former BLUFOR group is now an OPFOR group with an empty id
 - The former BLUFOR group takes "Alpha-1"
 - The original OPFOR group now has an empty id

Personally I'm not a fan of that scenario, since it is modifying a different group. I think the check is worth keeping, especially since it is relatively quick and infrequently used. I'm also thinking I'll add a new module to set a group ID from Zeus. I know this stuff is kinda niche but I think it'll be useful enough to add.

Reply:
So setting to an already used id (on the same side) removes the id from the group that already had it? I'd say it's worth to keep the check in that case. The new group should be empty, not the old one that wasn't even touched by the module.
Alternatively some way to add (2) etc. to the id, though idk if you think that is worth the trouble.

Reply:
I don't think adding (2) is worth it, either you don't care and will leave it empty, or you do care and are going to change it anyway.